lib/rules/no-unused-vars.js

```diff
@@ -155,7 +162,14 @@ module.exports = {
          * @returns {UnusedVarMessageData} The message data to be used with this unused variable.
          */
         function getAssignedMessageData(unusedVar) {
-            const additional = config.varsIgnorePattern ? `. Allowed unused vars must match ${config.varsIgnorePattern.toString()}` : "";
+            const def = unusedVar.defs && unusedVar.defs[0];
```

Comment:
```suggestion
            const def = unusedVar.defs[0];
```

`Variable#defs` always exists. If the variable doesn't have definitions, it will be an empty array.